pkg/proto/auth/auth.proto

```diff
@@ -27,3 +29,73 @@ message AuthenticationMetadata {
   // be accessed by JMESPathExpressionAuthorizer.
   google.protobuf.Value private = 3;
 }
+
+service Authentication {
+  // Authenticates a request that the client has received.
+  rpc Authenticate(AuthenticateRequest) returns (AuthenticateResponse) {}
+}
+
+message AuthenticateRequest {
+  message ValueList {
+    repeated string value = 1;
+  }
+
+  // The metadata is forwarded as part of the request as accessing the
+  // authentication service might have its own authentication and authorizing
+  // system.
+  map<string, ValueList> request_metadata = 1;
+
+  // A description of for example what subsystem is being accessed and what
+  // kind of request is being made.
+  google.protobuf.Value scope = 2;
+}
+
+message AuthenticateResponse {
+  oneof verdict {
+    // Allow the request.
+    //
+    // The value of this field is used as authentication metadata in
+    // literal form.
+    AuthenticationMetadata allow = 1;
+
+    // Deny the request, returning a fixed error message back
+    // to the client.
+    string deny = 2;
```

Comment:
Maybe good to document what kind of status code should be sent back to the client? Should it always be the equivalent of `Unauthenticated`? Same with `AuthorizeResponse.deny`.